### DISCUSSION TITLE
Set Target Rate 'Unknown Error Occurred' -

### DISCUSSION THREAD

**Discussion by lazy-pete** [BottlecapDave/HomeAssistant-OctopusEnergy#685] [Q&A]
Hi - I've been labouring to learn how to leverage all the power of HA and Octopus / solar / batteries etc. Your Octopus integration seems to be the way to get sensors detecting the optimal periods to charge / discharge batteries when it's not sunny, but I am having difficulty setting them up. The integration seems to be setup correctly, in that it shows three devices (meters) and 55 entities (a few disabled by the integration itself I believe. However, when I use the setup link through docs to create the Target Rate Period sensor I just get an 'Unknown Error Occurred' message. Screenshot attached. Any ideas whats going on ? Where would I find log files to support troubleshooting ? Apologies for the newbie questions. Ultimately trying to use these sensors to feed https://github.com/jevburchell/Solis-Modbus-Auto-Charging-using-Solax so the whole thing is automated !

![Screenshot 2024-01-07 20 33 00](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/30950467/240f123c-b77f-4405-9cad-44e72a26c50e)

**Comment by BottlecapDave** [comment]
Hello and sorry you're seeing this. Log files are located in the settings -> system -> logs. However as you have the integration installed I would suggest starting debug mode (see last item in faq) and then try again. My guess would be it failed to get the account information to validate the data, as there's very little else that can go wrong.

**Reply by lazy-pete** [reply]
Had another bash - same response. Appending log files in case someone can make sense of them - I think towards the bottom it says something about an error with account info, as you suggested. Any hints as to what I may have done wrong ? I presume the top entry in the UI is just a title, not a username or something ? 

[home-assistant_octopus_energy_2024-01-09T23-17-31.233Z.log](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/files/13881265/home-assistant_octopus_energy_2024-01-09T23-17-31.233Z.log)
[home-assistant_octopus_energy_2024-01-09T23-19-31.679Z.log](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/files/13881266/home-assistant_octopus_energy_2024-01-09T23-19-31.679Z.log)
<img width="596" alt="Screenshot 2024-01-09 at 23 25 30" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/30950467/8adc4032-cd8a-43f1-94fd-7b9b7cda4a72">

**Reply by lazy-pete** [reply]
After that, I remade my API, worked out the time format was wrong (no seconds needed) and the UI told me I needed lowercase naming. After several attempts it then finally worked !

**Reply by BottlecapDave** [reply]
Thanks for updating. Based on the screenshot it looks like you're on an older version of the integration which has had some additional validation added. I also have it on my list to update this screen to use the local cached version of the account instead of get the up to date version to minimize the impact when the API is being unresponsive.

**Comment by lazy-pete** [comment]
Thanks Dave - I’ll have another go and report back.

P
